test/core/ping.spec.js

```diff
@@ -0,0 +1,220 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const pull = require('pull-stream/pull')
+const drain = require('pull-stream/sinks/drain')
+const pullCatch = require('pull-catch')
+const parallel = require('async/parallel')
+const DaemonFactory = require('ipfsd-ctl')
+const isNode = require('detect-node')
+
+const expect = chai.expect
+chai.use(dirtyChai)
+const df = DaemonFactory.create({ exec: 'src/cli/bin.js' })
+
+const config = {
+  Bootstrap: [],
+  Discovery: {
+    MDNS: {
+      Enabled:
+        false
+    }
+  }
+}
+
+function spawnNode ({ dht = false }, cb) {
+  const args = dht ? ['--enable-dht-experiment'] : []
+  df.spawn({
+    args,
+    config,
+    initOptions: { bits: 512 }
+  }, cb)
+}
+
+describe('ping', function () {
+  this.timeout(60 * 1000)
+
+  if (!isNode) return
+
+  describe('DHT disabled', function () {
+    // Without DHT nodes need to be previously connected
+    let ipfsdA
+    let ipfsdB
+    let bMultiaddr
+    let ipfsdBId
+
+    // Spawn nodes
+    before(function (done) {
+      this.timeout(60 * 1000)
+
+      parallel([
+        spawnNode.bind(null, { dht: false }),
+        spawnNode.bind(null, { dht: false })
+      ], (err, ipfsd) => {
+        expect(err).to.not.exist()
+        ipfsdA = ipfsd[0]
+        ipfsdB = ipfsd[1]
+        done()
+      })
+    })
+
+    // Get the peer info object
+    before(function (done) {
+      this.timeout(60 * 1000)
+
+      ipfsdB.api.id((err, peerInfo) => {
+        expect(err).to.not.exist()
+        ipfsdBId = peerInfo.id
+        bMultiaddr = peerInfo.addresses[0]
+        done()
+      })
+    })
+
+    // Connect the nodes
+    before(function (done) {
+      this.timeout(60 * 1000)
+      ipfsdA.api.swarm.connect(bMultiaddr, done)
+    })
+
+    after((done) => ipfsdA.stop(done))
+    after((done) => ipfsdB.stop(done))
+
+    it('sends the specified number of packets', (done) => {
+      let packetNum = 0
+      const count = 3
+      pull(
+        ipfsdA.api.pingPullStream(ipfsdBId, { count }),
+        pullCatch(err => {
+          expect(err).to.not.exist()
+        }),
+        drain(({ Success, Time, Text }) => {
```

Comment:
This would either require the HTTP API to be changed (and consequently be different from the Go implementation) or the `js-ipfs-api` module to lowercase the responses (the later looks like a nicer option tbh). I'm not sure how desirable this would be, taking interoperability with `go-ipfs`  and such into account. @diasdavid any thoughts?

Reply:
I believe the current president is for js-ipfs core and js-ipfs-api to be `lowerCamelCase` but js-ipfs http to be compatible with Go i.e. `UpperCamelCase`.

It means that the js-ipfs http layer needs to translate from `lowerCamelCase` to `UpperCamelCase` ([example](https://github.com/ipfs/js-ipfs/blob/427f72b46e5bc5fd275a555f3939b449c357e1af/src/http/api/resources/files.js#L268-L270)) and js-ipfs-api needs to translate responses from `UpperCamelCase` to `lowerCamelCase` ([example](https://github.com/ipfs/js-ipfs-api/blob/master/src/utils/file-result-stream-converter.js)).

Reply:
@JGAntunes if you're busy I'm happy to finish this off on your behalf? You want to push what you've done?

Reply:
@alanshaw sure thing, feel free to pick it up if you have some time. I hope I'm not blocking anyone or anything with this. I'm away from my laptop, but tbh I hadn't done much, besides looking at similar stuff like you pointed out, copying some pieces of code and doing an initial skeleton of a transform stream that processed the ping response (all of this in `js-ipfs-api`). I'm counting on having some time to deal with this later this evening, but please do pick it up if you have a chance, just let me know :+1: